plugins/widget/plugin.js

```diff
@@ -2910,13 +2931,20 @@
 				return;
 
 			var toBeDowncasted = downcastingSessions[ evt.data.downcastingSessionId ],
-				toBe, widget, widgetElement, retElement, editableElement, e;
+				toBe, widget, widgetElement, retElement, editableElement, e, parserFragment;
 
 			while ( ( toBe = toBeDowncasted.shift() ) ) {
 				widget = toBe.widget;
 				widgetElement = toBe.element;
 				retElement = widget._.downcastFn && widget._.downcastFn.call( widget, widgetElement );
 
+				// In case of copying widgets, we replace the widget with clipboard data (#3138).
+				if ( evt.data.widgetsCopy && widget.getClipboardHtml ) {
+					parserFragment = CKEDITOR.htmlParser.fragment.fromHtml( widget.getClipboardHtml() );
+
+					retElement = parserFragment.children[ 0 ];
+				}
```

Comment:
I wonder how it will work with the code below, managing editables output 🤔 Does the fact that custom widget HTML (`getClipboardHtml()`) is provided means any editables will be simply replaced by this custom HTML? If so there is no need to process them in any way. WDYT?

And second thing, I wonder how it will (or should) work with nested widgets? Can you check it and also add such test (we don't have many nested widget cases but still worth checking)?

Reply:
Everything seems to work correctly. Editables and nested widgets are replaced by the outermost widget's `getClipboardHtml` result.

Reply:
Shouldn't we mentioned in the `getClipboardHtml()` docs that for nested widget it will overwrite/replace entire widget HTML (no matter nesting/editables content)? Or is it obvious? 🤔